lib/chutney/TorNet.py

```diff
@@ -1300,6 +1327,7 @@ def getNodeDirInfoStatusPattern(self, dir_format):
            the ed25519 key. (Or RSA block matching, which is hard.)
         """
         nickname = self.getNick()
+        #ed25519_id = self._setEd25519Id()
 
```

Comment:
@nmathewson I know this is a "#ed25519_id = self._setEd25519Id()" comment, but I did uncomment and this the wrong way to call a function, because I am using the the _setEd22519() and while reading the log files, I realized the function I am calling is creating the problem since the function is within the class. Can you suggest me improvements that I should do in the code to not let the test fail?

Reply:
So, reading the function you have added: It looks like this function doesn't return anything, so using its return value in this way will just get you "None".  It looks like it tries to set a field, however.  Perhaps you should look at the value of that field after this function is run?

Reply:
@nmathewson How should I look value of that field after this function is run?

Reply:
In _setEd22519(), you did:
`self._env['ed25519_id'] = ed25519_id`

These classes share the _env object, so all you need to do is:
```
ed25519_id = None
# old tor versions do not have an ed25519-id
if 'ed25519_id' in self._env:
    ed25519_id = self._env['ed25519_id']
```